tools/examples/src/util.ts

```diff
@@ -0,0 +1,85 @@
+/**
+ * @license
+ * Copyright 2019 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { promises as fs } from 'fs';
+import { ExampleMetadata } from './metadata';
+import { EXAMPLES_ROOT } from './main';
+import { relative, extname } from 'path';
+
+/**
+ * Reads a provided template file, transforms its content
+ * by calling a provided transform function,
+ * and then writes it to the file-system at the provided path.
+ */
+export async function transformAndWriteTemplate(
+  transformFn: (source: string) => string,
+  templateFile: string,
+  outFile: string,
+): Promise<string> {
+  let source = (await fs.readFile(templateFile, {
+    encoding: 'utf-8',
+  })).toString();
+
+  source = transformFn(source);
+
+  await fs.writeFile(outFile, source, {
+    flag: 'w', // "w" -> Create file if it does not exist
+    encoding: 'utf8',
+  });
+
+  return Promise.resolve(outFile);
```

Comment:
you don't need a promise resolve just return outFile – when a function is annotaed as async it will be a promise